### PR TITLE
bringing hash, address, and chain config into lib

### PR DESCRIPTION
**Pull request by hexoscott**

### CODE REVIEW

**Review by AskAlexSharov — APPROVED** [review]

File: common/bytes.go [hunk 1]
```diff
@@ -59,3 +62,42 @@ func EnsureEnoughSize(in []byte, size int) []byte {
 	}
 	return in[:size] // Reuse the space if it has enough capacity
 }
+
+func FromHex(s string) []byte {
```
Comment (yperbasis) [inline common/bytes.go:66]: These functions should be moved to `package hexutility`

File: common/blocktype/hash.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
+package blocktype
```
Comment (yperbasis) [inline common/blocktype/hash.go:1]: Why not leave it in `package common`?

File: common/bytes.go (hunk 1 above)
Comment (AskAlexSharov) [reply common/bytes.go:66]: There is some hex package: https://github.com/ledgerwatch/erigon-lib/tree/main/common/hex

**Review by AlexeyAkhunov — APPROVED** [review]

**Comment by hexoscott** [comment]
Hi @yperbasis, @AskAlexSharov,

Thank you both for the feedback.  I've made the changes requested.  The common package clashes with the one local to erigon itself, so i've aliased it to `libcommon` on the erigon side to give the distinction where there wasn't one already.

**Review by yperbasis — APPROVED** [review]

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**